github/provider_utils.go

```diff
@@ -0,0 +1,189 @@
+package github
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+)
+
+var testUser string = os.Getenv("GITHUB_TEST_USER")
+var testCollaborator string = os.Getenv("GITHUB_TEST_COLLABORATOR")
+var isEnterprise string = os.Getenv("ENTERPRISE_ACCOUNT")
+var testOrganization string = os.Getenv("GITHUB_ORGANIZATION")
+var testOwner string = os.Getenv("GITHUB_OWNER")
+var testToken string = os.Getenv("GITHUB_TOKEN")
+
+var testTokenGHES string = os.Getenv("GHES_TOKEN")
+var testBaseURLGHES string = os.Getenv("GHES_BASE_URL")
+
+func testAccPreCheckEnvironment(t *testing.T, requiredEnvironmentVariables []string) {
+	for _, variable := range requiredEnvironmentVariables {
+		if v := os.Getenv(variable); v == "" {
+			t.Fatal(variable + " must be set for acceptance tests")
+		}
+	}
+}
+
+func testAccPreCheck(t *testing.T) {
+	if v := os.Getenv("GITHUB_TOKEN"); v == "" {
+		t.Fatal("GITHUB_TOKEN must be set for acceptance tests")
+	}
+	if v := os.Getenv("GITHUB_OWNER"); v == "" {
+		t.Fatal("GITHUB_OWNER must be set for acceptance tests")
+	}
+	if v := os.Getenv("GITHUB_TEST_USER"); v == "" {
+		t.Fatal("GITHUB_TEST_USER must be set for acceptance tests")
+	}
+	if v := os.Getenv("GITHUB_TEST_COLLABORATOR"); v == "" {
+		t.Fatal("GITHUB_TEST_COLLABORATOR must be set for acceptance tests")
+	}
+	if v := os.Getenv("GITHUB_TEMPLATE_REPOSITORY"); v == "" {
+		t.Fatal("GITHUB_TEMPLATE_REPOSITORY must be set for acceptance tests")
+	}
+	if v := os.Getenv("GITHUB_TEMPLATE_REPOSITORY_RELEASE_ID"); v == "" {
+		t.Fatal("GITHUB_TEMPLATE_REPOSITORY_RELEASE_ID must be set for acceptance tests")
+	}
+}
+
+func testAccCheckOrganization() error {
+	baseURL := os.Getenv("GITHUB_BASE_URL")
+	token := os.Getenv("GITHUB_TOKEN")
+
+	config := Config{
+		BaseURL: baseURL,
+		Token:   token,
+		Owner:   testOwner,
```

Comment:
we'll need to check for either `testOwner` or `testOrganization` before passing one or the other to this config. thinking about this more, I believe this function  can be refactored in the future to be used in a `PreCheck` step and use the `testAccProvider` object that gets populated instead of having to create  a new config. for now, though, I think we can just check the environment variables to not skip tests when the organization is exported

Reply:
> check for either testOwner or testOrganization before passing one or the other to this config

this has been addressed with a recent commit.